pages/content/amp-dev/documentation/guides-and-tutorials/develop/custom-javascript.md

```diff
@@ -3,120 +3,143 @@ $title: Use custom JavaScript in AMP pages
 
-For user experience and security reasons, `amp-script` enforces DOM manipulation restrictions.
+The following are a few excellent use cases for `amp-script`.
 
-### User interaction
+### Create new interactions
 
-When a user interacts with elements wrapped inside an `<amp-script>` component, your custom JavaScript must return DOM manipulations quickly when needed. By default, changes to the DOM are permitted **less than one second** from the initial interaction. A notable exception is when your code must retrieve data from the network via `fetch`. Here DOM changes can be requested after the response is returned to the user and for **less than one second** afterward. If a script mutates the DOM outside of a permitted window, this will result in a fatal error and the `<amp-script>` component will terminate the Web Worker. A terminated `<amp-script>` component will not run again.
+When possible, it's better to use an AMP component for a DOM interaction. If the interaction you wish to create would be of interest to other developers, you are encouraged to [suggest and contribute](https://github.com/ampproject/amphtml/blob/master/CONTRIBUTING.md) a new component. Otherwise, `amp-script` provides a fine way to create interactions not currently supported by AMP components.
 
-### Unprompted changes
+### Add advanced logic
 
-There is no user interaction required to manipulate the DOM if the `<amp-script>` component has a fixed height.
+`amp-bind` allows you to introduce logic into user interactions, but your JavaScript needs to fit into a single expression. Because the code is encapsulated in an HTML attribute, it won't get the benefit of syntax highlighting in your IDE, you can't set breakpoints, and it may be hard to debug. For any logic that doesn't fit neatly into a compact expression, `amp-script` presents a promising alternative.
 
-## Script size
+### Enhance AMP components <a name="enhance-amp-components"></a>
 
-AMP enforces a limit of 150 kilobytes of custom JavaScript on each page. This limit is shared among all `<amp-script>` components on that page. Any external JavaScript library must be imported to each individual `<amp-script>` component.
+`amp-script` has access to AMP state variables and the `AMP.getState()` and `AMP.setState()` methods. This provides a path to enhance existing AMP components with your own logic. It also makes it possible to affect the DOM outside of the `<amp-script>` component itself. See [here](https://amp.dev/documentation/examples/components/amp-script/#interacting-with-%3Camp-state%3E) and [here](https://amp.dev/documentation/examples/components/amp-script/#interacting-with-amp-components) for examples.
 
-## Scope
+### Replace amp-bind and amp-list
```

Comment:
This conflicts with other guidance we are giving - mostly the first one in this section under "Create new interactions" when we advise devs to use components first.

Reply:
good catch. making this more consistent.